src/app/modules/shared_models/model_utils.nim

```diff
@@ -0,0 +1,20 @@
+import std/macros
+
+# Macro that simplifies checking and updating values in a model
+# IMPORTANT:
+  # The model's items need to be in a `seq` called `items`
+  # A `seq[string]` named `roles` needs to exist
+  # The index of the item being checked must be named `ind`
+macro updateRole*(propertyName: untyped, roleName: untyped): untyped =
```

Comment:
Hmm, can't we just pass the container as a reference into this function (e.g. `container` instead of `self.items`) and assert the container is of type `seq`? Sorry lacking the NIM-fu here :) Same for the `ind` index and `seq of roles`?

Reply:
Making it a macro means we can use in all types of models.

basically, the macro is executed during compilation and it inserts the code where the macro is called, so that way you don't have to pass `self`, `items`, `roles` or anything. They just have to exist in the context.

Since it's done at compile, it just won,t compile if the needed properties don't exist.